skglm/estimators.py

```diff
@@ -141,15 +143,26 @@ def _glm_fit(X, y, model, datafit, penalty):
     coefs, p_obj, kkt = solver(
         X_, y, datafit_jit, penalty_jit, w, Xw, max_iter=model.max_iter,
         max_epochs=model.max_epochs, p0=model.p0,
-        tol=model.tol,  # ws_strategy=model.ws_strategy,
+        tol=model.tol, fit_intercept=model.fit_intercept,
+        # ws_strategy=model.ws_strategy,
         verbose=model.verbose)
+    if y.ndim == 1:
+        model.coef_, model.stop_crit_ = coefs[:n_features], kkt
+        if model.fit_intercept:
+            model.intercept_ = coefs[-1]
+        else:
+            model.intercept_ = 0.
+    else:
+        model.coef_, model.stop_crit_ = coefs[:n_features, :], kkt
+        if model.fit_intercept:
+            model.intercept_ = coefs[-1, :]
+        else:
+            model.intercept_ = np.zeros(y.shape[1])
 
-    model.coef_, model.stop_crit_ = coefs, kkt
     model.n_iter_ = len(p_obj)
-    model.intercept_ = 0.
 
     if is_classif and n_classes_ <= 2:
-        model.coef_ = coefs[np.newaxis, :]
+        model.coef_ = coefs[np.newaxis, :n_features]
```

Comment:
Use None instead of np.newaxis  (btw they are the same object)

Reply:
Is there any reason to prefer `None` over `np.newaxis` ? `np.newaxis` seems more explicit to me https://stackoverflow.com/questions/944863/numpy-should-i-use-newaxis-or-none

Reply:
You're right, np.newaxis is clearer and there are no room constraints here